### PR TITLE
Consolidate profiler changes

### PR DESCRIPTION
**Pull request by tiennou**
This is a hodge-podge of changes, so here's a list:
- switch to GitHub Actions instead of Travis
- merge #28, #29 & #30
- add the remaining game classes added since, with some care that we don't error on missing ones (looking at you sim)
- make the callgrind filename from #29 contain the shard name so the tick isn't meaningless
- removes the `AlreadyWrappedException` in favor of merely repointing the wrapped function profiler object if it's already a wrapper
- log on profiler start, or empty profile
- throw on invalid registration instead as soon as possible
- add some tests which bring coverage up to 86%

### CODE REVIEW

**Comment by tiennou** [comment]
@gdborton Would you be okay with merging that, or otherwise clarifying the repo's status w.r.t maintainership?

File: .travis.yml [hunk 1]
```diff
@@ -1,6 +0,0 @@
-language: node_js
-node_js:
-  - '4'
-  - '6'
-script:
-  - 'npm test && cat ./coverage/lcov.info | ./node_modules/coveralls/bin/coveralls.js'
```
Comment (admon84) [inline .travis.yml:6]: Can the `coveralls` package be removed from package.json now?

**Review by admon84 — APPROVED** [review]
I tested this on my local pserver and everything seems to work well. Just one comment/question

**Comment by tiennou** [comment]
Thanks for the review and test. And yeah, I tested and coveralls doesn't need to be installed anymore. CI will handle it

**Pull request merged**